graphx/src/main/scala/org/apache/spark/graphx/Pregel.scala

```diff
@@ -123,16 +127,25 @@ object Pregel extends Logging {
       s" but got ${maxIterations}")
 
     var g = graph.mapVertices((vid, vdata) => vprog(vid, vdata, initialMsg)).cache()
+    val graphCheckpointer = new PeriodicGraphCheckpointer[VD, ED](
+      checkpointInterval, graph.vertices.sparkContext)
+    graphCheckpointer.update(g)
+
     // compute the messages
-    var messages = GraphXUtils.mapReduceTriplets(g, sendMsg, mergeMsg)
+    var messages = GraphXUtils.mapReduceTriplets(g, sendMsg, mergeMsg).cache()
+    val messageCheckpointer = new PeriodicRDDCheckpointer[(VertexId, A)](
+      checkpointInterval, graph.vertices.sparkContext)
+    messageCheckpointer.update(messages.asInstanceOf[RDD[(VertexId, A)]])
```

Comment:
Actually, for the sake of simplicity and consistency, I'm going to suggest we keep the checkpointer update calls but remove all `.cache()` calls. The `update` calls persist the underlying data, making the calls to `.cache()` unnecessary.